repose-aggregator/src/docs/asciidoc/recipes/functional-test-framework.adoc

```diff
@@ -0,0 +1,163 @@
+= Functional Test Framework
+
+The Repose functional test framework is a testing library provided by the Repose team meant to simplify the process of creating functional tests for features in Repose or its extension components.
+
+This framework is based on the http://spockframework.org/[Spock framework].
+While there are features of this framework that are can be used without Spock, it is expected that tests utilizing this framework will be Spock tests.
+
+== Download
+
+The Repose functional test framework is published to https://maven.research.rackspacecloud.com/[an instance of Nexus hosted by Rackspace].
+
+It is highly recommended that a build system be used to manage this library.
+The following snippets provide the necessary additions to download this framework with some popular build systems.
+
+=== Maven
+
+[source,xml]
+.pom.xml
+----
+<repositories>
+    <repository>
+        <id>public.maven.research.rackspace.com</id>
+        <name>Rackspace Research</name>
+        <url>https://maven.research.rackspacecloud.com/content/groups/aggregate</url>
+    </repository>
+</repositories>
+
+<dependencies>
+    <dependency>
+      <groupId>org.openrepose</groupId>
+      <artifactId>functional-test-framework</artifactId>
+      <version>${version}</version>
+    </dependency>
+</dependencies>
+----
+
+=== Gradle
+
+[source,groovy]
+.build.gradle
+----
+repositories {
+    maven {
+        url "https://maven.research.rackspacecloud.com/content/groups/aggregate/"
+    }
+}
+
+dependencies {
+    compile "org.openrepose:functional-test-framework:${version}"
+}
+----
+
+== Setup
+
+There are two requirements that must be satisfied for this framework.
+
+The first requirement is that a `repose-test.properties` file is on the classpath when tests are run.
+The `repose-test.properties` file tells that framework what version of Repose, directories, host, and ports to use.
+It should be in the https://en.wikipedia.org/wiki/.properties#Format[Java properties file format] like the following example:
+
+----
```

Comment:
Sure, I can do that.

So using coderay, I don't see [explicit support for property files](http://coderay.rubychan.de/). As such, applying the `source` block style probably doesn't do anything (just checked, it doesn't). Actually, I'm slightly worried that just applying the `source` block style will potentially mess up this blocks highlighting if we ever set the `source-language` document attribute (see [here](http://asciidoctor.org/docs/user-manual/#applying-source-highlighting)).